### PR TITLE
Map voting weight.

### LINKED ISSUE #7508: Map Rotation Voting Compromise
As it stands, we very rarely play other maps, and while I wasn't in favour of the forced map rotation when it was around, due to both the complete lack of accountability of n3x and the lack of democracy, I accept there's an element of truth to N3x's argument and hence I propose that instead of absolutely voting for the map, you rather vote for for the weighting of the random chance of the map. - Here's a practical, albeit drastically oversimplified example.

Assume 40% of the players want to play Box, 15% want to play Defficiency, 30% want to play Metaclub, 15% want to play Taxi. If there was a vote, assuming they only voted for the map they wanted to play, the maps played would be thus: 100% Box station.

With the weighted system, instead you would have 40% Box rounds, 15% Defficiency rounds, 30% Metaclub rounds, & 15% Taxi rounds.

That's far fairer, as both a compromise and an alternative.

### PR DESCRIPTION
**Pull request by PJB3005**
Makes map voting weight based.

Fixes #7508 

**NOTE:** Due to VV being fucking terrible I'm unable to actually test this.

### CODE REVIEW

**Comment by ghost** [comment]
@PJB3005 add fixes #7508 #7370 #7418 #7353

**Comment by gbasood** [comment]
I don't get it but hey :+1:

**Comment by Exxion** [comment]
@DrCelt But all of those except the first are already fixed and closed

**Comment by ghost** [comment]
:+1:

File: code/modules/html_interface/voting/voting.dm [hunk 1]
```diff
@@ -28,24 +28,25 @@ var/global/datum/controller/vote/vote = new()
...
 	var/datum/html_interface/nanotrasen/vote/interface
 	var/list/data
 	var/list/status_data
-	var/last_update = 0
-	var/initialized = 0
-	var/lastupdate = 0
+	var/last_update    = 0
+	var/initialized    = 0
+	var/lastupdate     = 0
+	var/weighed        = FALSE // Whether to use weighed voting.
```
Comment (clusterfack) [inline code/modules/html_interface/voting/voting.dm]: Weighted

This is **critical**

Comment (PJB3005) [inline code/modules/html_interface/voting/voting.dm]: `:<`

**Comment by Zth--** [comment]
:+1:

**Pull request merged**